dask/config.py

```diff
@@ -102,6 +102,36 @@ def merge(*dicts):
     return result
 
 
+def normalize_key(key):
+    """ Replaces underscores with hyphens
+
+    Parameters
+    ----------
+    key : string
+        Key to assign.
+    """
+    return key.replace('_', '-')
+
+
+def normalize_nested_keys(config):
+    """ Replaces underscores with hyphens for keys for a nested Mapping
+
+    Examples
+    --------
+    >>> a = {'x': 1, 'y_1': {'a_2': 2}}
+    >>> normalize_nested_keys(a)
+    {'x': 1, 'y-1': {'a-2': 2}}
+    """
+    config_norm = {}
+    for key, value in config.items():
+        if isinstance(value, Mapping):
+            value = normalize_nested_keys(value)
+        key_norm = normalize_key(key)
```

Comment:
It looks like `{1: 2}` is valid yaml so not all keys are strings.

Reply:
I don't know when this would come up, but it's not entirely out of scope.

Reply:
Thanks for the review, I added a check to only call the `replace` method if the key is a string